kogito-benchmarks-framework/src/main/java/org/kie/kogito/benchmarks/framework/Commands.java

```diff
@@ -0,0 +1,484 @@
+        List<String> quarkusNativeProperties = System.getProperties().entrySet().stream()
+                .filter(e -> e.getKey().toString().contains("quarkus.native"))
+                .map(e -> "-D" + e.getKey() + "=" + e.getValue())
+                .collect(Collectors.toList());
+        return quarkusNativeProperties;
+    }
+
+    public static String getQuarkusPlatformVersion() {
+        for (String p : new String[] { "QUARKUS_PLATFORM_VERSION", "quarkus.platform.version" }) {
+            String env = System.getenv().get(p);
+            if (StringUtils.isNotBlank(env)) {
+                return env;
+            }
+            String sys = System.getProperty(p);
+            if (StringUtils.isNotBlank(sys)) {
+                return sys;
+            }
+        }
+        logger.warn("Failed to detect quarkus.platform.version/QUARKUS_PLATFORM_VERSION, defaulting to getQuarkusVersion().");
+        return getQuarkusVersion();
+    }
+
+    public static String getQuarkusVersion() {
+        for (String p : new String[] { "QUARKUS_VERSION", "quarkus.version" }) {
+            String env = System.getenv().get(p);
+            if (StringUtils.isNotBlank(env)) {
+                return env;
+            }
+            String sys = System.getProperty(p);
+            if (StringUtils.isNotBlank(sys)) {
+                return sys;
+            }
+        }
+        String failure = "Failed to determine quarkus.version. Check pom.xm, check env and sys vars QUARKUS_VERSION";
+        try (Scanner sc = new Scanner(new File(getBaseDir() + File.separator + "pom.xml"), StandardCharsets.UTF_8)) {
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                Matcher m = quarkusVersionPattern.matcher(line);
+                if (m.matches()) {
+                    return m.group(1);
+                }
+            }
+        } catch (IOException e) {
+            throw new IllegalArgumentException(failure);
+        }
+        throw new IllegalArgumentException(failure);
+    }
+
+    public static String getBaseDir() {
+        String baseDirValue = getSystemPropertyOrEnvVarValue("basedir");
+        return new File(baseDirValue).getParent();
+    }
+
+    public static String getAppsDir() {
+        return getSystemPropertyOrEnvVarValue("appsDir");
+    }
+
+    private static String getSystemPropertyOrEnvVarValue(String name) {
+        String systemPropertyValue = System.getProperty(name);
+        if (StringUtils.isNotBlank(systemPropertyValue)) {
+            return systemPropertyValue;
+        }
+
+        String envPropertyValue = System.getenv(name);
+
+        if (StringUtils.isBlank(envPropertyValue)) {
+            throw new IllegalArgumentException("Unable to determine the value of the property " + name);
+        }
+        return envPropertyValue;
+    }
+
+    public static void cleanTarget(App app) {
+        String target = APPS_DIR + File.separator + app.dir + File.separator + "target";
```

Comment:
I think we're using these in `cleanDirOrFile` as input to `Paths.get()`, wouldn't it be better to use Path methods instead of string concat here?

Reply:
Good catch, implemented.